src/main/java/data/plans/PlanList.java

```diff
@@ -140,22 +140,27 @@ public Plan createAndAddPlan(String userArgument) throws ArrayIndexOutOfBoundsEx
     /**
      * Checks if the provided plan details already exists in the ArrayList of plans. A plan
      * is considered to already exist in the list if the plan name matches an existing plan
-     * name in the ArrayList.
+     * name in the ArrayList. Additionally, checks if the plan name is called "rest day".
+     * If it is, do not allow a plan called "rest day" as it is used in the schedule feature.
      *
      * @param userPlanNameInput The plan name entered by user to check.
      * @return True if an existing plan with the same plan name exists in the plans list.
      *         Otherwise, returns false.
      */
-    public boolean checkForExistingPlanName(String userPlanNameInput) {
+    public boolean checkPlanNameValidity(String userPlanNameInput) {
         String userPlanNameInputLowerCase = userPlanNameInput.toLowerCase();
+        if (userPlanNameInputLowerCase.equals("rest day")) {
```

Comment:
Should we use a constant for `"rest day"`?

Reply:
oh yes thank you for this!